lib/postProcess.js

```diff
@@ -37,10 +38,22 @@ postProcess._fetchRelatedResources = function(request, mainResource, callback) {
   var dataItems = mainResource[request.params.relation];
   if (!(dataItems instanceof Array)) dataItems = [ dataItems ];
 
-  var resourcesToFetch = dataItems.map(function(dataItem) {
-    return jsonApi._apiConfig.pathPrefix + dataItem.type + "/" + dataItem.id;
+  var resourcesToFetch = dataItems.reduce(function(map, dataItem) {
+    map[dataItem.type] = map[dataItem.type] || [ ];
+    map[dataItem.type].push(dataItem.id);
```

Comment:
```javascript?
if (dataItem.type in map) {
  map[dataItem.type].push(dataItem.id);
}
```
instead of `42-43`? Same effect but improved readability?